### ISSUE TITLE
ExceptionDispatchInfo.SetCurrentStackTrace doesn't include method in stack trace in WASM

### ISSUE THREAD

**Issue by safern** [dotnet/runtime#39341]
```cs
Exception e;

 e = new Exception();
ABCDEFGHIJKLMNOPQRSTUVWXYZ(e);
Assert.Contains(nameof(ABCDEFGHIJKLMNOPQRSTUVWXYZ), e.StackTrace);

[MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]
private static void ABCDEFGHIJKLMNOPQRSTUVWXYZ(Exception e)
{
    Assert.Same(e, ExceptionDispatchInfo.SetCurrentStackTrace(e));
}
```

This would fail in `Assert.Contains` because the `StackTrace` only contains this line: 
https://github.com/dotnet/runtime/blob/b032a4d9044bafca6d805b8f1ed244e0400c45de/src/mono/netcore/System.Private.CoreLib/src/System/Exception.Mono.cs#L118

It seems like this line is effectively returning empty:
https://github.com/dotnet/runtime/blob/b032a4d9044bafca6d805b8f1ed244e0400c45de/src/mono/netcore/System.Private.CoreLib/src/System/Exception.Mono.cs#L117

cc: @steveisok @marek-safar @akoeplinger

**Comment by akoeplinger** [comment]
Looks like a variation of https://github.com/dotnet/runtime/issues/39223

**Comment by lewing** [comment]
Fixed in #40031
